File: src/main/java/com/elchologamer/userlogin/util/LogFilter.java
package com.elchologamer.userlogin.util;

import com.elchologamer.userlogin.UserLogin;
import org.apache.commons.lang.StringUtils;
import org.apache.logging.log4j.Level;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Marker;
import org.apache.logging.log4j.core.Filter;
import org.apache.logging.log4j.core.LogEvent;
import org.apache.logging.log4j.core.Logger;
import org.apache.logging.log4j.message.Message;
import org.bukkit.configuration.ConfigurationSection;

import java.util.ArrayList;
import java.util.List;

public class LogFilter implements Filter {

    public static void register() {
        Logger logger = ((Logger) LogManager.getRootLogger());
        logger.addFilter(new LogFilter());
    }

    private Result filter(String message) {

        if (!StringUtils.containsIgnoreCase(message, " issued server command: /")) return Result.NEUTRAL;

        List<String> commands = new ArrayList<>();
        commands.add("login");
        commands.add("register");
        commands.add("changepassword");

        ConfigurationSection section = UserLogin.getPlugin().getConfig().getConfigurationSection("commandAliases");
        if (section != null) {
            commands.addAll(section.getStringList("login"));
            commands.addAll(section.getStringList("register"));
            commands.addAll(section.getStringList("changepassword"));
        }

        for (String command : commands) {
            if (StringUtils.containsIgnoreCase(message, command)) {
                return Result.DENY;
            }
        }

        return Result.NEUTRAL;
    }

    @Override
    public Result getOnMismatch() {
        return Result.NEUTRAL;
    }

    @Override
    public Result getOnMatch() {
        return Result.NEUTRAL;
    }

    @Override
    public Result filter(Logger logger, Level level, Marker marker, String s, Object... objects) {
        return filter(s);
    }

    @Override
    public Result filter(Logger logger, Level level, Marker marker, String s, Object o) {
        return filter(s);
    }

    @Override
    public Result filter(Logger logger, Level level, Marker marker, String s, Object o, Object o1) {
        return filter(s);
    }

    @Override
    public Result filter(Logger logger, Level level, Marker marker, String s, Object o, Object o1, Object o2) {
        return filter(s);
    }

    @Override
    public Result filter(Logger logger, Level level, Marker marker, String s, Object o, Object o1, Object o2, Object o3) {
        return filter(s);
    }

    @Override
    public Result filter(Logger logger, Level level, Marker marker, String s, Object o, Object o1, Object o2, Object o3, Object o4) {
        return filter(s);
    }

    @Override
    public Result filter(Logger logger, Level level, Marker marker, String s, Object o, Object o1, Object o2, Object o3, Object o4, Object o5) {
        return filter(s);
    }

    @Override
    public Result filter(Logger logger, Level level, Marker marker, String s, Object o, Object o1, Object o2, Object o3, Object o4, Object o5, Object o6) {
        return filter(s);
    }

    @Override
    public Result filter(Logger logger, Level level, Marker marker, String s, Object o, Object o1, Object o2, Object o3, Object o4, Object o5, Object o6, Object o7) {
        return filter(s);
    }

    @Override
    public Result filter(Logger logger, Level level, Marker marker, String s, Object o, Object o1, Object o2, Object o3, Object o4, Object o5, Object o6, Object o7, Object o8) {
        return filter(s);
    }

    @Override
    public Result filter(Logger logger, Level level, Marker marker, String s, Object o, Object o1, Object o2, Object o3, Object o4, Object o5, Object o6, Object o7, Object o8, Object o9) {
        return filter(s);
    }

    @Override
    public Result filter(Logger logger, Level level, Marker marker, Object o, Throwable throwable) {
        return filter(o.toString());
    }

    @Override
    public Result filter(Logger logger, Level level, Marker marker, Message message, Throwable throwable) {
        return filter(message.getFormattedMessage());
    }

    @Override
    public Result filter(LogEvent e) {
        return filter(e.getMessage().getFormattedMessage());
    }

    @Override
    public State getState() {
        return State.STARTED;
    }

    @Override
    public void initialize() {
    }

    @Override
    public void start() {
    }

    @Override
    public void stop() {
    }

    @Override
    public boolean isStarted() {
        return true;
    }

    @Override
    public boolean isStopped() {
        return false;
    }
}

File: src/main/kotlin/com/elchologamer/userlogin/util/LogFilter.kt
package com.elchologamer.userlogin.util

import org.apache.logging.log4j.Level
import org.apache.logging.log4j.LogManager
import org.apache.logging.log4j.Marker
import org.apache.logging.log4j.core.Filter
import org.apache.logging.log4j.core.LifeCycle
import org.apache.logging.log4j.core.LogEvent
import org.apache.logging.log4j.core.Logger
import org.apache.logging.log4j.message.Message

internal object LogFilter : Filter {

    fun register() {
        val logger = LogManager.getRootLogger() as Logger
        logger.addFilter(this)
    }

    private fun filter(message: String): Filter.Result {
        return if (message.matches(Regex("(.+) issued server command: /(?i)(login|register|changepassword)(.*)"))) {
            Filter.Result.DENY
        } else {
            Filter.Result.NEUTRAL
        }
    }

    override fun getOnMismatch() = Filter.Result.NEUTRAL
    override fun getOnMatch() = Filter.Result.NEUTRAL
    override fun filter(logger: Logger, level: Level, marker: Marker, s: String, vararg objects: Any) = filter(s)
    override fun filter(logger: Logger, level: Level, marker: Marker, s: String, o: Any) = filter(s)
    override fun filter(logger: Logger, level: Level, marker: Marker, s: String, o: Any, o1: Any) = filter(s)

    override fun filter(
        logger: Logger,
        level: Level,
        marker: Marker,
        s: String,
        o: Any,
        o1: Any,
        o2: Any
    ) = filter(s)

    override fun filter(
        logger: Logger,
        level: Level,
        marker: Marker,
        s: String,
        o: Any,
        o1: Any,
        o2: Any,
        o3: Any
    ) = filter(s)

    override fun filter(
        logger: Logger,
        level: Level,
        marker: Marker,
        s: String,
        o: Any,
        o1: Any,
        o2: Any,
        o3: Any,
        o4: Any
    ) = filter(s)

    override fun filter(
        logger: Logger,
        level: Level,
        marker: Marker,
        s: String,
        o: Any,
        o1: Any,
        o2: Any,
        o3: Any,
        o4: Any,
        o5: Any
    ) = filter(s)

    override fun filter(
        logger: Logger,
        level: Level,
        marker: Marker,
        s: String,
        o: Any,
        o1: Any,
        o2: Any,
        o3: Any,
        o4: Any,
        o5: Any,
        o6: Any
    ) = filter(s)

    override fun filter(
        logger: Logger,
        level: Level,
        marker: Marker,
        s: String,
        o: Any,
        o1: Any,
        o2: Any,
        o3: Any,
        o4: Any,
        o5: Any,
        o6: Any,
        o7: Any
    ) = filter(s)

    override fun filter(
        logger: Logger,
        level: Level,
        marker: Marker,
        s: String,
        o: Any,
        o1: Any,
        o2: Any,
        o3: Any,
        o4: Any,
        o5: Any,
        o6: Any,
        o7: Any,
        o8: Any
    ) = filter(s)

    override fun filter(
        logger: Logger,
        level: Level,
        marker: Marker,
        s: String,
        o: Any,
        o1: Any,
        o2: Any,
        o3: Any,
        o4: Any,
        o5: Any,
        o6: Any,
        o7: Any,
        o8: Any,
        o9: Any
    ) = filter(s)

    override fun filter(logger: Logger, level: Level, marker: Marker, o: Any, throwable: Throwable) =
        filter(o.toString())

    override fun filter(
        logger: Logger,
        level: Level,
        marker: Marker,
        message: Message,
        throwable: Throwable
    ) = filter(message.formattedMessage)

    override fun filter(e: LogEvent): Filter.Result = filter(e.message.formattedMessage)

    override fun getState() = LifeCycle.State.STARTED

    override fun initialize() {}

    override fun start() {}
    override fun stop() {}

    override fun isStarted() = true
    override fun isStopped() = false
}